firestore/src/main/composite_filter_main.cc

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#if defined(__ANDROID__)
+#error "This header should not be used on Android."
+#endif
+
+#include "firestore/src/main/composite_filter_main.h"
+#include "Firestore/core/src/core/composite_filter.h"
+
+namespace firebase {
+namespace firestore {
+
+template <typename... Ts>
+CompositeFilterInternal::CompositeFilterInternal(
+    core::CompositeFilter::Operator op, const Filter& filter, Ts... filters)
+    : FilterInternal(FilterType::Composite),
+      op_(op),
+      filters_(filter, filters...) {}
```

Comment:
`filters_` is a vector of shared pointers. I'm pretty sure that the unpacking (`filters...`) will copy the `FilterInternal` values into that vector, but we should check this. (i.e. I'm worried that the shared pointers point to somewhere in the stack that'll be garbage after `filters...` goes out of scope).

Reply:
uh.. wait, how is this initializing from Filter to shared_ptr<FilterInternal> ?

Reply:
I completely reworked this, passing a vector of filters to internal classes instead.